database/misc/mongodb_replset.py

```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2015, Jakub Jirutka <jakub@jirutka.cz>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: mongodb_replset
+author: Jakub Jirutka
+version_added: "1.8"
+short_description: Initiate or extend replica set.
+description:
+  - This module allows to initiate a new replica set or add new members to an
+    existing one. It can't remove members from a replica set, this should be
+    always done with caution by hand.
+options:
+  login_user:
+    description:
+      - The username used to authenticate with.
+    aliases: [ "user" ]
+    required: false
+  login_password:
+    description:
+      - The password used to authenticate with.
+    aliases: [ "password" ]
+    required: false
+  login_host:
+    description:
+      - The first host to connect when initiating the replica set.
+    aliases: [ "host" ]
+    required: false
+    default: localhost
+  login_port:
+    description:
+      - Port of the first host to connect when initiating the replica set.
+    aliases: [ "port" ]
+    required: false
+    default: "27017"
+  hosts:
+    description:
+      - A comma delimited list of replica set members.
+    aliases: [ "members" ]
+    required: true
+    example: mongo0,mango1:27011,mango2:27012
+  replica_set:
+    description:
+      - Name of the replica set to create or connect to.
+    aliases: [ "replset" ]
+    required: true
+    default: "rs0"
+'''
+
+import ConfigParser
+try:
+    from pymongo.errors import ConnectionFailure, OperationFailure, ConfigurationError
+    from pymongo import MongoClient
+    from pymongo.mongo_replica_set_client import MongoReplicaSetClient
+    from pymongo.read_preferences import ReadPreference
+    from pymongo.uri_parser import split_hosts
+except ImportError:
+    pymongo = None
+
+
+def load_mongocnf_creds():
+    """ Read credentials from ~/.mongodb.cnf file, when exists. """
+    config = ConfigParser.RawConfigParser()
+    mongocnf = os.path.expanduser('~/.mongodb.cnf')
+
+    try:
+        config.readfp(open(mongocnf))
```

Comment:
Fixed. I’ve refactored code to use [RawConfigParser.read](https://docs.python.org/2.6/library/configparser.html#ConfigParser.RawConfigParser.read) (instead of readfp) that takes care about closing the file itself.